frontend/src/components/createPostForm/createPostForm.js

```diff
@@ -1,43 +1,68 @@
-import React, { useState } from 'react';
+import React, {useState} from 'react';
+// import {useForm} from "react-hook-form";
+// import app from "../app/App";
 
 const CreatePostForm = ({callback}) => {
+    // const {register,handleSubmit:handleFormSubmitReactHookForm} = useForm()
     const [textArea, setTextArea] = useState("");
-
+    const [photoFile, setPhotoFile] = useState("");
     const handleTextAreaChange = (event) => {
         setTextArea(event.target.value);
     }
+    const handlePhotoFileChange = (event) => {
+        console.log(event.target.files[0])
+        setPhotoFile(event.target.files[0]);
+
+    }
 
     const handleSubmit = async (event) => {
+        // a try catch block for this function would allow for better Error
+        // for this function I have refactored out the separate request and send the whole post in formData
         event.preventDefault();
-        if (textArea !== '') {
-            let response = await fetch('/posts', {
-                method: 'post',
-                headers: {
-                    'Authorization': "Bearer " + window.localStorage.getItem('token'),
-                    'Content-Type': 'application/json',
-                },
-                body: JSON.stringify({ user_id: window.localStorage.getItem('user_id'), message: textArea})
-            })
-
-            const data = await response.json()
-
-            if(response.status !== 201) {
-                console.log(data.error)
-            } else {
-                console.log("post created");
-                setTextArea('');
-                callback(true);
-            }
+        // console.log(photoFile)
+        // array_buffer []
+
+        let token = window.localStorage.getItem('token')
+        const formData = new FormData();
+        formData.append('user_id', window.localStorage.getItem('user_id'));
+
+        if (photoFile) formData.append('photoFile', photoFile);
+
+        if (textArea) formData.append('message', textArea);
+
+        const response = await fetch('/posts', {
+            method: 'post',
+            headers: {
+                'Authorization': `Bearer ${token}`
+            },
+            body: formData,
+        });
+        const data = await response.json();
+        if (response.status !== 201) {
+            console.log(data.error);
+        } else {
+            console.log('post created');
+            setTextArea('');
+            setPhotoFile(null);
+            callback(true);
         }
+
     }
```

Comment:
Is this right? Did we get the photo upload working?

Reply:
No, I think Charlie and I started working from Murat's code in the hopes that it would be done before EOD! We can solve in the resolve conflicts editor